sunpy/physics/differential_rotation.py

```diff
@@ -87,317 +78,73 @@ def diff_rot(duration, latitude, rot_type='howard', frame_time='sidereal'):
     if frame_time == 'synodic':
         rotation_deg -= 0.9856 * delta_days
 
-    #return Longitude((np.round(rotation_deg, 4)), u.deg)
-    return np.round(rotation_deg, 4) * u.deg
-
+    return rotation_deg * u.deg
 
-@u.quantity_input(x=u.arcsec, y=u.arcsec)
-def rot_hpc(x, y, tstart, tend, frame_time='synodic', rot_type='howard', **kwargs):
-    """Given a location on the Sun referred to using the Helioprojective
-    Cartesian co-ordinate system (typically quoted in the units of arcseconds)
-    use the solar rotation profile to find that location at some later or
-    earlier time.  Note that this function assumes that the data was observed
-    from the Earth or near Earth vicinity.  Specifically, data from SOHO and
-    STEREO observatories are not supported.  Note also that the function does
-    NOT use solar B0 and L0 values provided in source FITS files - these
-    quantities are calculated.
 
+def solar_rotate_coordinate(coordinate,
+                            new_observer_time,
+                            new_observer_location="earth", **diff_rot_kwargs):
+    """Given a coordinate on the Sun, calculate where that coordinate maps to
+    at some later or earlier time, given the solar rotation profile.  Note that
+    if the new observer location is defined using a BaseCoordinateFrame or
+    SkyCoord, then it is assumed that the new observer location is correct for
+    the new observer time that was also passed in.
     Parameters
     ----------
-    x : `~astropy.units.Quantity`
-        Helio-projective x-co-ordinate in arcseconds (can be an array).
-
-    y : `~astropy.units.Quantity`
-        Helio-projective y-co-ordinate in arcseconds (can be an array).
-
-    tstart : `sunpy.time.time`
-        date/time to which x and y are referred.
-
-    tend : `sunpy.time.time`
-    date/time at which x and y will be rotated to.
-
-    rot_type : {'howard' | 'snodgrass' | 'allen'}
-        | howard: Use values for small magnetic features from Howard et al.
-        | snodgrass: Use Values from Snodgrass et. al
-        | allen: Use values from Allen, Astrophysical Quantities, and simpler
-          equation.
-
-    frame_time : {'sidereal' | 'synodic'}
-        Choose type of day time reference frame.
-
+    coordinate : `~sunpy.coordinates`
```

Comment:
type should just be `astropy.coordinates.SkyCoord` the description should say something like "any valid coordinate which is transformable to heliographic stonyhurst.